library/src/main/java/org/xmtp/android/library/frames/FramesClient.kt

```diff
@@ -0,0 +1,94 @@
+package org.xmtp.android.library.frames
+
+import android.util.Base64
+import org.xmtp.android.library.Client
+import org.xmtp.android.library.XMTPException
+import org.xmtp.android.library.frames.FramesConstants.PROTOCOL_VERSION
+import org.xmtp.android.library.messages.PrivateKeyBuilder
+import org.xmtp.android.library.messages.Signature
+import org.xmtp.android.library.messages.getPublicKeyBundle
+import org.xmtp.proto.message.contents.PublicKeyOuterClass.SignedPublicKeyBundle
+import java.security.MessageDigest
+import org.xmtp.proto.message.contents.Frames.FrameActionBody
+import org.xmtp.proto.message.contents.Frames.FrameAction
+import java.util.Date
+
+class FramesClient(private val xmtpClient: Client, var proxy: OpenFramesProxy = OpenFramesProxy()) {
+
+    suspend fun signFrameAction(inputs: FrameActionInputs): FramePostPayload {
+        val opaqueConversationIdentifier = buildOpaqueIdentifier(inputs)
+        val frameUrl = inputs.frameUrl
+        val buttonIndex = inputs.buttonIndex
+        val inputText = inputs.inputText
+        val state = inputs.state
+        val now = Date().time * 1_000_000
+        val frameActionBuilder = FrameActionBody.newBuilder().also { frame ->
+            frame.frameUrl = frameUrl
+            frame.buttonIndex = buttonIndex
+            frame.opaqueConversationIdentifier = opaqueConversationIdentifier
+            frame.timestamp = now
+            frame.unixTimestamp = now.toInt()
+            if (inputText != null) {
+                frame.inputText = inputText
+            }
+            if (state != null) {
+                frame.state = state
+            }
```

Comment:
Is there a reason you null check here? Seems like state and inputText are okay with null being set?

```suggestion
                frame.inputText = inputText
                frame.state = state
```

Reply:
They are not, it errors if the state is null or inputText is null, instead they use the default values

Reply:
Ahh okay now I see you did this in iOS

`let state = inputs.state ?? ""`
 Then to clean this up what about this